### PR TITLE
Feature/add json support

### LINKED ISSUE #2063: Support parsing for AWS CloudFormation JSON files
### Description
Currently, we do not support the CloudFormation JSON files, only YAML. The goal is to provide our parsing engine the ability to parse JSON files from CloudFormation while maintaining existing query support

### PR DESCRIPTION
**Pull request by joaoReigota1**
Closes #2063 

**Proposed Changes**

- add JSON samples and JSON support to cloud formation
- fixed bug with config file
- fixed bug when scanning files instead of dirs
- Increased timeout to 10 seconds
I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/cloudFormation/alb_is_not_integrated_with_waf/test/positive4.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "Resources": {
+    "MyLoadBalancerV2": {
+      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
+      "Properties": {
+        "Name": "myloadbalancerv2",
+        "Scheme": "internet-facing"
+      }
+    }
+  }
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/alb_is_not_integrated_with_waf/test/positive4.json:12]: please check EOF newline missing for json samples

File: internal/console/scan.go [hunk 2]
```diff
@@ -55,7 +56,7 @@ var scanCmd = &cobra.Command{
 
 func initializeConfig(cmd *cobra.Command) error {
 	if cfgFile == "" {
-		_, err := os.Stat(filepath.ToSlash(filepath.Join(path, "kics.config")))
+		_, err := os.Stat(filepath.ToSlash(filepath.Join(filepath.Dir(path), "kics.config")))
```
Comment (rogeriopeixotocx) [inline internal/console/scan.go:59]: related with json support or just slipped right in?

File: pkg/parser/json/parser.go [hunk 3]
```diff
@@ -31,5 +31,5 @@ func (p *Parser) GetKind() model.FileKind {
 
 // SupportedTypes returns types supported by this parser, which are cloudFormation
 func (p *Parser) SupportedTypes() []string {
-	return []string{"JSON"}
+	return []string{"CloudFormation"}
```
Comment (rogeriopeixotocx) [inline pkg/parser/json/parser.go:34]: nice

File: test/main_test.go [hunk 4]
```diff
@@ -49,19 +50,23 @@ func TestMain(m *testing.M) {
...
-	files, err := filepath.Glob(path.Join(q.dir, fmt.Sprintf(filePattern, strings.ToLower(string(q.kind)))))
-	x0 := filepath.FromSlash(path.Join(q.dir, "test/positive_expected_result.json"))
-	for i, check := range files {
-		if check == x0 {
-			files = append(files[:i], files[i+1:]...)
+	var files []string
+	for _, kinds := range q.kind {
+		kindFiles, err := filepath.Glob(path.Join(q.dir, fmt.Sprintf(filePattern, strings.ToLower(string(kinds)))))
+		x0 := filepath.FromSlash(path.Join(q.dir, "test/positive_expected_result.json"))
+		for i, check := range kindFiles {
```
Comment (rogeriopeixotocx) [inline test/main_test.go:62]: this is all just for skipping `test/positive_expected_result.json` right?

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
minor changes requested

File: test/main_test.go (hunk 4 above)
Comment (joaoReigota1) [reply test/main_test.go:62]: yes

File: internal/console/scan.go (hunk 2 above)
Comment (joaoReigota1) [reply internal/console/scan.go:59]: related to a bug with config file

**Review by ruigomescx — APPROVED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**